hazelcast/src/main/java/com/hazelcast/core/Endpoint.java

```diff
@@ -39,5 +41,14 @@
      *
      * @return the socket address for this endpoint
      */
+    @Deprecated
     SocketAddress getSocketAddress();
+
+    /**
+     * Returns the socket address for this endpoint for the given qualifier.
+     *
+     * @return the socket address for this endpoint for the given qualifier.
+     */
+    //TODO (TK) : Clients dont have qualifiers ; so maybe separate it
+    SocketAddress getSocketAddress(EndpointQualifier qualifier);
```

Comment:
The API will be removed, just had a discussion with @vbekiaris on this topic, and this is not really needed, since one can get the same info through the Member class directly.